### PR TITLE
Update workspace CLI examples

### PR DESCRIPTION
**Pull request by denniseik**
# PR into Azure/azureml-examples

## Checklist

I have:

- [ ] read and followed the contributing guidelines

## Changes

- Update files names from 'basic' to 'workspace'. Someone may manage a simple YAML spec to manage a complex workspace configuration, but only update selected properties.
- Include friendly name and make clear difference between resource name and friendly name by example.
- Remove private link examples
- Include comments to show purpose of examples

fixes #

### CODE REVIEW

File: cli/assets/workspace/workspace-hbi.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This configuration shows how to configure a workspace with the hbi flag enabled.
+# This flag specifies whether to reduce telemetry collection and enable additional encryption when high-business-impact data is used.
+$schema: https://azuremlschemas.azureedge.net/latest/workspace.schema.json
+name: mlw-hbiexample-prd
+location: eastus
+friendly_name: HBI-workspace example
+description: Use this field to describe the purpose of the workspace.
```
Comment (lostmygithubaccount) [inline cli/assets/workspace/workspace-hbi.yml:7]: same as above

File: cli/assets/workspace/workspace-minimal.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# This configuration shows the minimal set of properties required to manage a workspace using YML
+$schema: https://azuremlschemas.azureedge.net/latest/workspace.schema.json
```
Comment (lostmygithubaccount) [inline cli/assets/workspace/workspace-minimal.yml:2]: remove for minimal example

**Review by lostmygithubaccount — CHANGES_REQUESTED** [review]
in general, please:

- don't include "workspace" in the filename - it's already in the path, the entire directory is workspaces
- follow the naming convention for other resources, assets, job names - ending with " -example" after other descriptive words separated by hyphens
- remove comments (possibly move to descriptions)
- remove tags, have a single tag example
-

File: cli/assets/workspace/workspace-cmk.yml [hunk 3]
```diff
@@ -1,6 +1,9 @@
+# This configurations shows how to create a workspace that uses customer-managed keys for encryption.
 $schema: https://azuremlschemas.azureedge.net/latest/workspace.schema.json
-name: cmk-example
+name: mlw-cmkexample-prd
```
[the remark this answers is not in the document]
Comment (denniseik) [reply cli/assets/workspace/workspace-cmk.yml:3]: This recommendation is inconsistent with Azure abbrevation guidance https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

File: cli/assets/workspace/workspace-cmk.yml [hunk 4]
```diff
@@ -1,6 +1,9 @@
+# This configurations shows how to create a workspace that uses customer-managed keys for encryption.
```
[the remark this answers is not in the document]
Comment (denniseik) [reply cli/assets/workspace/workspace-cmk.yml:1]: People are likely to open azureml-examples repo directly from github and don't have any context on the YML spec nor docs...

File: cli/assets/workspace/workspace-minimal.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
+name: mlw-minimalexample-prd
```
[the remark this answers is not in the document]
Comment (denniseik) [reply cli/assets/workspace/workspace-minimal.yml:3]: I think we need to follow this guidance for consistency with ARM/Terraform and rest of Azure infra examples per https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations

File: cli/assets/workspace/workspace-cmk.yml [hunk 6]
```diff
@@ -1,6 +1,9 @@
...
 $schema: https://azuremlschemas.azureedge.net/latest/workspace.schema.json
-name: cmk-example
+name: mlw-cmkexample-prd
 location: eastus
+friendly_name: CMK-workspace example
+description: Use this field to describe the purpose of the workspace.
 customer_managed_key: 
   key_vault: /subscriptions/"<YOUR_SUBSCRIPTION_ID>"/resourceGroups/"<YOUR_RESOURCE_GROUP>"/providers/Microsoft.KeyVault/vaults/"<YOUR_KEY_VAULT>"
```
[the remark this answers is not in the document]
Comment (denniseik) [reply cli/assets/workspace/workspace-cmk.yml:8]: Good spot!

**Pull request merged**